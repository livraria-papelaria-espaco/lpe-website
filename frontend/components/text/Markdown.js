import { Link, Typography } from '@material-ui/core';
import { withStyles } from '@material-ui/core/styles';
import ReactMarkdown from 'markdown-to-jsx';
import React from 'react';

const styles = (theme) => ({
  listItem: {
    marginTop: theme.spacing(1),
  },
});

const markdownOptions = {
  overrides: {
    h1: {
      component: Typography,
      props: {
        gutterBottom: true,
        variant: 'h5',
      },
    },
    h2: { component: Typography, props: { gutterBottom: true, variant: 'h6' } },
    h3: { component: Typography, props: { gutterBottom: true, variant: 'subtitle1' } },
    h4: {
      component: Typography,
      props: { gutterBottom: true, variant: 'caption', paragraph: true },
    },
    p: { component: Typography, props: { paragraph: true } },
    a: { component: Link },
    li: {
      component: withStyles(styles)(({ classes, ...props }) => (
        <li className={classes.listItem}>
          <Typography component='span' {...props} />
        </li>
      )),
    },
  },
};

const Markdown = (props) => <ReactMarkdown options={markdownOptions} {...props} />;

export default Markdown;
