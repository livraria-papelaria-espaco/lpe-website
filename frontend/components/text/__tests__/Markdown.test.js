import React from 'react';
import { mount } from 'enzyme';
import Markdown from '../Markdown';

describe('Markdown renderer (/components/text/Markdown)', () => {
  it('Renders some markdown correctly when no custom className is provided', () => {
    const text = `
# This is a heading 1
## This is a heading 2

Some text  
and a line break

[This is a link](https://google.com)

![And an image](https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png)

Also renders **bold** and _italic_ correctly.
`;

    const wrapper = mount(<Markdown>{text}</Markdown>);

    expect(wrapper.html()).toMatchSnapshot();
  });

  it('Renders some markdown correctly when a custom className is provided', () => {
    const text = `Just a paragraph`;

    const wrapper = mount(<Markdown className='someClassName'>{text}</Markdown>);

    expect(wrapper.html()).toMatchSnapshot();
  });
});
