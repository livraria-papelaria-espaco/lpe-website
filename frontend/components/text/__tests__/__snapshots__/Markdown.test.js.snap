// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown renderer (/components/text/Markdown) Renders some markdown correctly when a custom className is provided 1`] = `"<span class=\\"someClassName\\">Just a paragraph</span>"`;

exports[`Markdown renderer (/components/text/Markdown) Renders some markdown correctly when no custom className is provided 1`] = `"<div class=\\"\\"><h5 class=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom\\" id=\\"this-is-a-heading-1\\">This is a heading 1</h5><h6 class=\\"MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom\\" id=\\"this-is-a-heading-2\\">This is a heading 2</h6><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\">Some text<br>and a line break</p><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\"><a class=\\"MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary\\" href=\\"https://google.com\\">This is a link</a></p><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\"><img src=\\"https://www.google.com/images/branding/googlelogo/1x/googlelogo_color_272x92dp.png\\" alt=\\"And an image\\" class=\\"Component-img-38 undefined\\"></p><p class=\\"MuiTypography-root MuiTypography-body1 MuiTypography-paragraph\\">Also renders <strong>bold</strong> and <em>italic</em> correctly.</p></div>"`;
